### PR TITLE
irc: Clean up config options (fixes #113)

### LINKED ISSUE #113: Allow user to configure more variables for IRC server connection (e.g. port, SSL certs, etc.)
# Summary

Allow user to configure more parts of the IRC server connection (e.g. server port, using self-signed TLS certs, etc.)

# Background

IRC servers are a wild world. Everyone can do things a little bit differently or deploy their network in a different way. Most of the current developers run the bot on Freenode, so we haven't thought about other bots as much. However, to make the bot more useful for more people, more options should be opened up for configuration for those who need it.

# Details

Implementation should be straightforward. New env variables should be created for the following:

* IRC server port
* Allow expired / self-signed TLS certificates

These env variables should be used when negotiating the connection to the IRC server, instead of using hard-coded defaults. For our purposes, we can ship the Freenode defaults in our `env.example` file.

# Outcome

More people can use the bot on a range of different IRC networks, outside of only Freenode

---

_Original comment_:

There is a way to specify IRC server port as well as SSL self signed certificate or turning on debug in the .env file or as environment variables?

### PR DESCRIPTION
**Pull request by jwflory**
This commit reworks the IRC-side configuration options to support new
parameters and configuration options:

* **IRC_BOT_NAME**: IRC bot nick. Already existed, but now better
  documentation alongside other confusingly-named variables.
* **IRC_BOT_REALNAME**: `REALNAME` field support. Allows freeform text.
  Added bot advertisement as a default option if left unset.
* **IRC_BOT_IDENT**: Seldom used, but occasionally useful ident field.
  Overridden if NickServ credentials provided.
* **IRC_NICKSERV_USER**: Explicitly set a NickServ username for a bot.
  Previously used the `IRC_BOT_NAME`, but those are not always identical
  in all situations.

This commit also reworks the `env.example` file to better organize all
IRC options, in addition to adding the new ones. I also expanded and
reworked the config file glossary in the docs.

This Pull Request does not implement tests, but I would like to ask to
follow up on this in a subsequent Pull Request, preferably during the
upcoming testing sprint.

Fixes #113.

### CODE REVIEW

**Comment by jwflory** [comment]
Oh, I was also able to test these changes locally. It worked as expected for me. 👍🏻

**Review by robbyoconnor — APPROVED** [review]

**Comment by robbyoconnor** [comment]
(Just closed to trigger Travis)

**Comment by jwflory** [comment]
Travis CI came back with errors, should be fixed in commit 4bbc74ec1555d96b8e0c63293091e4328ea0ddd7. Once the test results come back, I will merge the Pull Request.

**Comment by robbyoconnor** [comment]
Tests are failing

**Comment by jwflory** [comment]
Commit 38e67f76cf47b12f2d753a028105f1022d398712 fixes the tests. Waiting for Travis CI results to come back, will check again in the morning. The same steps in Travis worked for me locally this time on the last commit.

**Comment by robbyoconnor** [comment]
@jwflory if they pass, I'll merge

**Pull request merged**
